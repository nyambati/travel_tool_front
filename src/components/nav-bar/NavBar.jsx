import React, { PureComponent } from 'react';

import PropTypes from 'prop-types';

import logo from '../../images/logo.svg';
import icon from '../../images/icon.svg';
import notification from '../../images/notification.svg';
import SearchBar from '../search-bar/SearchBar';
import './NavBar.scss';

/**
 * @description - Contains SearhBar componentents, Logo, user avatar and logout
 *
 * @class NavBar
 *
 * @extends {PureComponent}
 *
 */

class NavBar extends PureComponent {
  renderLogo() {
    return (
      <span className="logo-icons">
        <img src={logo} alt="Andela Logo" />
      </span>
    );
  }

  renderNotification() {
    const { onNotificationToggle } = this.props;
    return (
      <div
        id="notification"
        onClick={onNotificationToggle}
        className="nav-size"
        role="presentation"
      >
        <span className="material-icons mdl-badge" data-badge="12">
          <img
            src={notification}
            alt="Notification"
            className="navbar-notification"
            data-badge="10"
          />
        </span>
      </div>
    );
  }

  renderUserIcons() {
    const { avatar } = this.props;
    return (
      <div>
        <span className="mdl-icons">
          <img src={avatar} alt="Andela Logo" className="mdl-upic" />
          <span className="text-size">
            Silm Momoh
          </span>
        </span>
        <span>
          <button type="button" id="demo-menu-lower-right" className="mdl-button mdl-js-button mdl-button--icon mdl-Icons">
            <img src={icon} alt="Dropdown Icon" className="mdl-Icon" />
          </button>
          <div className="mdl-list">
            <ul htmlFor="demo-menu-lower-right" className="mdl-menu mdl-menu--bottom-right mdl-js-menu mdl-js-ripple-effect">
              <li className="mdl-menu__item">
                Logout
              </li>
            </ul>
          </div>
        </span>
      </div>
    );
  }
  render() {
    return (
      <header className="mdl-layout__header">
        <div className="mdl-layout__header-row">
          <div className="nav-size logo-icons">
            {this.renderLogo()}
          </div>
          <div className="mdl-layout-spacer" />
          <div className="search-size">
            <SearchBar />
          </div>
          <nav className="mdl-navigation">
            {this.renderNotification()}
            <div className="user-icon nav-size">
              {this.renderUserIcons()}
            </div>
          </nav>
        </div>
      </header>
    );
  }
}

NavBar.propTypes = {
  avatar: PropTypes.string.isRequired,
  onNotificationToggle: PropTypes.func.isRequired
};

export default NavBar;
