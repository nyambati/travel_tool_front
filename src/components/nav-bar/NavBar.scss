/* move this to header */
.mdl-layout__header {
  height: 78px;
  width: 1440px;
  background-color: #ffffff;
}

.logo-icons {
  height: 30px;
  width: 100px;
  margin-top: 15px;
}

.nav-size {
  margin-right: 25px;
}
.search-size {
  margin-right: -40px;
}
.navbar-notification {
  height: 21.13px;
  width: 16.25px;
  margin-top: 15px;
  color: gray;
}

.user-icon {
  height: 32px;
  width: 161px;
  margin-right: 150px;
  margin-top: 15px;
}

.text-size {
  margin: 7px;
  color: #4f4f4f;
  font-family: "DIN Pro";
  font-size: 14px;
  height: 21px;
  width: 90px;
  line-height: 21px;
}

.navbar-dropdown {
  color: gray;
}

.mdl-badge.mdl-badge:after {
  box-sizing: border-box;
  top: 15px;
  right: -10px;
  height: 17px;
  width: 17px;
  border: 1px solid #ffffff;
  background-color: #0459e4;
  color: #ffffff;
  font-family: "DIN Pro";
  font-size: 10px;
  font-weight: 500;
  letter-spacing: -0.13px;
  line-height: 12px;
  text-align: center;
}

.mdl-upic {
  box-sizing: border-box;
  height: 34px;
  width: 34px;
  border: 1px solid #ffffff;
  margin-right: 2px;
}

.mdl-Icons {
  margin-bottom: 5px;
}
.mdl-Icon {
  height: 4.99px;
  width: 8.99px;
}

.mdl-list {
  position: relative;
  top: -55px;
  left: 8px;
  margin: 0 2px;
}
.mdl-menu__item {
  height: 21px;
  width: 50px;
  color: #3359db;
  font-family: "DIN Pro";
  font-size: 16px;
  line-height: 21px;
}
.mdl-menu {
  width: 50px;
  height: 25px;
  border-radius: 2px;
  background-color: #ffffff;
}
