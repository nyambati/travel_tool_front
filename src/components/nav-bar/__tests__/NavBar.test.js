import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../NavBar';

// describe what we are testing
describe("Render NavBar component", () => {
  const props = {
    onNotificationToggle: jest.fn(),
    avatar: "avatar"
  };
  // make our assertions and what we expect to happen
  it("should match snapshot", () => {
    const wrapper = shallow(<NavBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the navbar as expected", () => {
    const wrapper = shallow(<NavBar {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it("should render the OnNotification click as exepected", () => {
    const wrapper = shallow(<NavBar {...props} />);
    let NotificationToggleSpy = jest.spyOn(
      wrapper.instance().props,
      "onNotificationToggle"
    );
    wrapper
      .find("#notification")
      .first()
      .simulate("click");
    expect(NotificationToggleSpy).toHaveBeenCalled();
  });
});
