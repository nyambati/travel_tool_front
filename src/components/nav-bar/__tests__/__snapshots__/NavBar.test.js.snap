// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar component should match snapshot 1`] = `
<header
  className="mdl-layout__header"
>
  <div
    className="mdl-layout__header-row"
  >
    <div
      className="nav-size logo-icons"
    >
      <span
        className="logo-icons"
      >
        <img
          alt="Andela Logo"
          src="logo.svg"
        />
      </span>
    </div>
    <div
      className="mdl-layout-spacer"
    />
    <div
      className="search-size"
    >
      <SearchBar />
    </div>
    <nav
      className="mdl-navigation"
    >
      <div
        className="nav-size"
        id="notification"
        onClick={[Function]}
        role="presentation"
      >
        <span
          className="material-icons mdl-badge"
          data-badge="12"
        >
          <img
            alt="Notification"
            className="navbar-notification"
            data-badge="10"
            src="notification.svg"
          />
        </span>
      </div>
      <div
        className="user-icon nav-size"
      >
        <div>
          <span
            className="mdl-icons"
          >
            <img
              alt="Andela Logo"
              className="mdl-upic"
              src="avatar"
            />
            <span
              className="text-size"
            >
              Silm Momoh
            </span>
          </span>
          <span>
            <button
              className="mdl-button mdl-js-button mdl-button--icon mdl-Icons"
              id="demo-menu-lower-right"
              type="button"
            >
              <img
                alt="Dropdown Icon"
                className="mdl-Icon"
                src="icon.svg"
              />
            </button>
            <div
              className="mdl-list"
            >
              <ul
                className="mdl-menu mdl-menu--bottom-right mdl-js-menu mdl-js-ripple-effect"
                htmlFor="demo-menu-lower-right"
              >
                <li
                  className="mdl-menu__item"
                >
                  Logout
                </li>
              </ul>
            </div>
          </span>
        </div>
      </div>
    </nav>
  </div>
</header>
`;
